SparseGrids/TasmanianSparseGrid.cpp

```diff
@@ -818,6 +848,31 @@ template<typename FloatType> const FloatType* TasmanianSparseGrid::formCanonical
 template const float* TasmanianSparseGrid::formCanonicalPoints(const float *x, Data2D<float> &x_temp, int num_x) const;
 template const double* TasmanianSparseGrid::formCanonicalPoints<double>(const double *x, Data2D<double> &x_temp, int num_x) const;
 
+template<typename FloatType> void TasmanianSparseGrid::diffCanonicalTransform(const FloatType*, FloatType *jacobian_diag) const {
+    int num_dimensions = base->getNumDimensions();
+    for(int i=0; i<num_dimensions; i++)
+        jacobian_diag[i] = 1.0;
+    if (conformal_asin_power.size() != 0) {
+        throw std::runtime_error("ERROR: in diffCanonicalTransform() derivatives/Jacobians are not available for conformal mappings");
+    }
+    if (domain_transform_a.size() != 0) {
+        // Based on the logic in mapTransformedToCanonical(). These are all affine transforms.
+        TypeOneDRule rule = base->getRule();
+        if (rule == rule_gausslaguerre or rule == rule_gausslaguerreodd)
+            for(int j=0; j<num_dimensions; j++)
+                jacobian_diag[j] *= (FloatType) domain_transform_b[j];
```

Comment:
You initialize to `1.0` then you multiply one by the transform and then reassign.

If you use just the assignment `=`, then you will skip the multiplication and one extra `1.0`.